mesonbuild/build.py

```diff
@@ -2405,119 +2438,6 @@ def get_transitive_build_target_deps(self) -> T.Set[T.Union[BuildTarget, 'Custom
-            if '@INPUT@' in i or '@INPUT0@' in i:
-                m = 'Output cannot contain @INPUT@ or @INPUT0@, did you ' \
-                    'mean @PLAINNAME@ or @BASENAME@?'
-                raise InvalidArguments(m)
-            # We already check this during substitution, but the error message
-            # will be unclear/confusing, so check it here.
-            if len(inputs) != 1 and ('@PLAINNAME@' in i or '@BASENAME@' in i):
-                m = "Output cannot contain @PLAINNAME@ or @BASENAME@ when " \
-                    "there is more than one input (we can't know which to use)"
-                raise InvalidArguments(m)
-        self.outputs = substitute_values(self.outputs, values)
-        if not self.name:
-            self.name = self.outputs[0]
-        self.capture = kwargs.get('capture', False)
-        if self.capture and len(self.outputs) != 1:
-            raise InvalidArguments('Capturing can only output to a single file.')
-        self.feed = kwargs.get('feed', False)
-        if self.feed and len(self.sources) != 1:
-            raise InvalidArguments('Feeding can only input from a single file.')
-        self.console = kwargs.get('console', False)
-        if not isinstance(self.console, bool):
-            raise InvalidArguments('"console" kwarg only accepts booleans')
-        if self.capture and self.console:
-            raise InvalidArguments("Can't both capture output and output to console")
-        if 'command' not in kwargs:
-            raise InvalidArguments('Missing keyword argument "command".')
-        if kwargs.get('depfile') is not None:
-            depfile = kwargs['depfile']
-            if not isinstance(depfile, str):
-                raise InvalidArguments('Depfile must be a string.')
-            if os.path.basename(depfile) != depfile:
-                raise InvalidArguments('Depfile must be a plain filename without a subdirectory.')
-            self.depfile = depfile
-        self.command = self.flatten_command(kwargs['command'])
-        for c in self.command:
-            if self.capture and isinstance(c, str) and '@OUTPUT@' in c:
-                raise InvalidArguments('@OUTPUT@ is not allowed when capturing output.')
-            if self.feed and isinstance(c, str) and '@INPUT@' in c:
-                raise InvalidArguments('@INPUT@ is not allowed when feeding input.')
-        if 'install' in kwargs:
-            self.install = kwargs['install']
-            if not isinstance(self.install, bool):
-                raise InvalidArguments('"install" must be boolean.')
-            if self.install:
-                if not kwargs.get('install_dir', False):
-                    raise InvalidArguments('"install_dir" must be specified '
-                                           'when installing a target')
-
-            if isinstance(kwargs['install_dir'], list):
-                FeatureNew.single_use('multiple install_dir for custom_target', '0.40.0', self.subproject)
```

Comment:
This warning is being totally dropped?

Reply:
And we can even add location now since we're handling this in the right place :)

Reply:
Indeed. :D